localization/localization_node/src/localization.cc

```diff
@@ -27,55 +27,60 @@
+  LOAD_PARAM(loc_params.add_best_previous_image, config, prefix);
+  LOAD_PARAM(loc_params.hamming_distance, config, prefix);
+  LOAD_PARAM(loc_params.goodness_ratio, config, prefix);
+  LOAD_PARAM(loc_params.use_clahe, config, prefix);
+  LOAD_PARAM(loc_params.num_extra_localization_db_images, config, prefix);
+  LOAD_PARAM(loc_params.verbose_localization, config, "");
+  LOAD_PARAM(loc_params.visualize_localization_matches, config, "");
+
+  // Detector Params
+  double min_threshold, default_threshold, max_threshold, goodness_ratio;
   int min_features, max_features, detection_retries;
-  double min_brisk_threshold, default_brisk_threshold, max_brisk_threshold;
-  camera::CameraParameters cam_params(config, "nav_cam");
-  if (!config->GetInt("num_similar", &num_similar))
-    ROS_FATAL("num_similar not specified in localization.");
-  if (!config->GetInt("ransac_inlier_tolerance", &ransac_inlier_tolerance))
-    ROS_FATAL("ransac_inlier_tolerance not specified in localization.");
-  if (!config->GetInt("ransac_iterations", &ransac_iterations))
-    ROS_FATAL("ransac_iterations not specified in localization.");
-  if (!config->GetInt("min_features", &min_features))
-    ROS_FATAL("min_features not specified in localization.");
-  if (!config->GetInt("max_features", &max_features))
-    ROS_FATAL("max_features not specified in localization.");
-  if (!config->GetInt("detection_retries", &detection_retries))
-    ROS_FATAL("detection_retries not specified in localization.");
-  if (!config->GetInt("histogram_equalization", &histogram_equalization))
-    ROS_FATAL("histogram_equalization not specified in localization.");
-
-  // For the brisk thresholds and other values, quietly assume some defaults
-  if (!config->GetReal("min_brisk_threshold", &min_brisk_threshold))
-    min_brisk_threshold = 20.0;
-  if (!config->GetReal("default_brisk_threshold", &default_brisk_threshold))
-    default_brisk_threshold = 90.0;
-  if (!config->GetReal("max_brisk_threshold", &max_brisk_threshold))
-    max_brisk_threshold = 110.0;
-  if (!config->GetInt("early_break_landmarks", &early_break_landmarks))
-    early_break_landmarks = 100;
+  LOAD_PARAM(min_threshold, config, prefix);
+  LOAD_PARAM(default_threshold, config, prefix);
+  LOAD_PARAM(max_threshold, config, prefix);
+  LOAD_PARAM(detection_retries, config, prefix);
+  LOAD_PARAM(min_features, config, prefix);
+  LOAD_PARAM(max_features, config, prefix);
 
   // This check must happen before the histogram_equalization flag is set into the map
   // to compare with what is there already.
   sparse_mapping::HistogramEqualizationCheck(map_->GetHistogramEqualization(),
-                                             histogram_equalization);
+                                             loc_params.histogram_equalization);
+  // Check consistency between clahe params
+  if (loc_params.use_clahe && (loc_params.histogram_equalization != 3 || map_->GetHistogramEqualization() != 3)) {
```

Comment:
The literal 3 here is a bit obscure. Probably there should be an enum or at least a macro that you can compare to?

Reply:
Done